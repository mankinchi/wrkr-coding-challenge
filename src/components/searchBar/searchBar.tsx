import { ChangeEventHandler } from "react";
import { Search } from "react-feather";

interface Props {
	value: string;
	onChange: (newValue: string) => void;
}

export const SearchBar = (props: Props) => {
	const { value, onChange } = props;

	const handleInputChange: ChangeEventHandler<HTMLInputElement> = (e) => {
		onChange(e.currentTarget.value);
	};

	return (
		<div className="rounded border border-solid border-slate-600 bg-slate-100 p-4">
			<div className="flex items-center gap-2 rounded-2xl border border-solid border-slate-600 bg-white px-3 py-2">
				<Search size={16} />
				<input
					className="grow focus:outline-none"
					type="text"
					value={value}
					onChange={handleInputChange}
					data-cy="searchbox"
				/>
			</div>
		</div>
	);
};
