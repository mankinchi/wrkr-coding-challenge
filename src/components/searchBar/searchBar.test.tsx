import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { SearchBar } from "./searchBar";

describe("Search Bar", () => {
	it("should render correctly", () => {
		const { container } = render(
			<SearchBar
				value="1"
				onChange={jest.fn()}
			/>,
		);

		expect(container).toMatchSnapshot();
	});

	it("should behave correctly", () => {
		const onChange = jest.fn();
		render(
			<SearchBar
				value="1"
				onChange={onChange}
			/>,
		);

		userEvent.type(screen.getByRole("textbox"), "2");

		expect(onChange).toHaveBeenCalledTimes(1);
		expect(onChange).toHaveBeenCalledWith("12");
	});
});
