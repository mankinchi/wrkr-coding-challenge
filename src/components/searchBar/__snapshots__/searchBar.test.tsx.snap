// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar should render correctly 1`] = `
<div>
  <div
    class="rounded border border-solid border-slate-600 bg-slate-100 p-4"
  >
    <div
      class="flex items-center gap-2 rounded-2xl border border-solid border-slate-600 bg-white px-3 py-2"
    >
      <svg
        fill="none"
        height="16"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <line
          x1="21"
          x2="16.65"
          y1="21"
          y2="16.65"
        />
      </svg>
      <input
        class="grow focus:outline-none"
        type="text"
        value="1"
      />
    </div>
  </div>
</div>
`;
